quickwit-storage/src/object_storage/azure_compatible_storage.rs

```diff
@@ -0,0 +1,508 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::num::NonZeroU32;
+use std::ops::Range;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use std::{fmt, io};
+
+use async_trait::async_trait;
+use azure_core::error::ErrorKind;
+use azure_core::{Pageable, StatusCode};
+use azure_storage::core::prelude::*;
+use azure_storage::Error as AzureError;
+use azure_storage_blobs::blob::operations::GetBlobResponse;
+use azure_storage_blobs::prelude::*;
+use bytes::Bytes;
+use futures::stream::StreamExt;
+use md5::Digest;
+use once_cell::sync::OnceCell;
+use quickwit_aws::retry::{retry, RetryParams, Retryable};
+use quickwit_common::{chunk_range, into_u64_range};
+use regex::Regex;
+use tantivy::directory::OwnedBytes;
+use thiserror::Error;
+use tokio::fs::File;
+use tokio::io::AsyncWriteExt;
+use tracing::instrument;
+
+use crate::{
+    MultiPartPolicy, PutPayload, Storage, StorageError, StorageErrorKind, StorageFactory,
+    StorageResolverError, StorageResult,
+};
+
+/// S3 Object storage Uri Resolver
+#[derive(Default)]
+pub struct AzureCompatibleBlobStorageFactory;
+
+impl StorageFactory for AzureCompatibleBlobStorageFactory {
+    fn protocol(&self) -> String {
+        "azure".to_string()
+    }
+
+    fn resolve(&self, uri: &str) -> Result<Arc<dyn Storage>, StorageResolverError> {
+        let storage = AzureCompatibleBlobStorage::from_uri(uri)?;
+        Ok(Arc::new(storage))
+    }
+}
+
+/// Azure Blob Storage compatible object storage implementation.
```

Comment:
I found "Azure Blob Storage compatible object storage implementation."
We could keep only "Azure object storage implementation."
(it's very, very minor).